### DISCUSSION TITLE
Image resize ext: Is there a way to get a file width and height after resize?

### DISCUSSION THREAD

**Discussion by dosstx** [firebase/extensions#1139] [General]
After resizing of image using the resize extension, is there a way to get the image's final width and height ?

I plan to use a dynamic image transformation service like Cloudinary to handle on the fly image transformation and resizing -- but I will use Firebase Storage as the media host. I will use the Firebase Image resizing extension to upload the image to storage and resize the original image to an efficient size for usage on Cloudinary (so users don't fill up my app's storage space quickly).  But, in order to successfully use Cloudinary, I need to get the image's originally resized image. Is it possible to get that meta data from this extension?

**Comment by yamankatby** [comment]
Hi @dosstx,
From your description, I think you are trying to reach the resized/generated images information. Of course, there is a way by using **Events**.

The extension can publish a resize completion event which you can optionally enable when you install the extension (or by updating it if it is already installed). If you enable events, you can [write custom event handlers](https://firebase.google.com/docs/extensions/install-extensions#eventarc) that respond to these events. The event includes information about the resized/generated images.

Event handler example:
```js
import { onCustomEventPublished } from "firebase-functions/v2/eventarc";
      
export const eventhandler = onCustomEventPublished(
    "firebase.extensions.storage-resize-images.v1.complete",
    (e) => {
        // Handle extension event here.
    });
```

**Reply by dosstx** [reply]
> Hi @dosstx, From your description, I think you are trying to reach the resized/generated images information. Of course, there is a way by using **Events**.

Hi @yamankatby, 
 Many thanks for taking the time to respond to my question. I have a follow up question as I don't think I was able to achieve what I wanted. 

Let me explain. First, I wrote the following function and deployed it. It did write a document in Firestore with the image's completion event:

`functions/getImageDimensions.js`:
```js
import { onCustomEventPublished } from 'firebase-functions/v2/eventarc'
import { logger } from 'firebase-functions/v1'
import { initializeApp } from 'firebase-admin/app'
import { getFirestore } from 'firebase-admin/firestore'

initializeApp()

// This extension publishes events to the projects/XXX/locations/us-central1/channels/firebase channel.

export const eventhandler = onCustomEventPublished(
  'firebase.extensions.storage-resize-images.v1.complete',
  (e) => {
    // Handle extension event here.
    logger.info('Received image resize completed event', e)
    // For example, write resized image details into Firestore.
    return getFirestore()
      .collection('testevents')
      .doc(e.subject.replace('/', '_')) // original file path
      .set(e)
    // .set(e.data) // resized images paths and sizes
  }
)
```

The above code creates the following Firestore document shown below:

![image](https://user-images.githubusercontent.com/6342102/185678937-2bc0fa14-6983-4d60-9c85-797105b3e5b8.png)

As you can see in the highlighted part of the above screenshot, the `size` property is `2000x2000`. I was hoping the `size` property would be the actual size of the resized image and not the resized parameter size from the extension config. **The size of the resized image is actually `1361x2000` and thus I was wanting to get that information to write to Firestore.**

Do you know if it's possible to get the image's actual size after resizing or will this require a custom cloud function to be created on my end? 

Thank you.

**Reply by iampava** [reply]
I have the exact some usecase. Any updates on this?
